### DISCUSSION TITLE
Passing extra config settings to the CIBW_BUILD_FRONTEND.

### DISCUSSION THREAD

**Discussion by andyfaff** [pypa/cibuildwheel#1222] [Q&A] [answered]
I would like to use the `CIBW_BUILD_FRONTEND: "build"`. More importantly I am interested in passing extra configuration settings to this front-end.

For example, if I use `build` directly I can pass extra config settings using the `C` flag:

```
python -m build . -C--opt=value
```

Is there a way of providing those extra config settings to `cibuildwheel`, such that it will then provide them to the `CIBW_BUILD_FRONTEND`? i.e. something along the lines of `CIBW_BUILD_FRONTEND_FLAGS`.

 I can't find anything in the documentation that would suggest that's possible.

**Comment by Czaki** [comment]
The cibuildwheel set `CIBUILDWHEEL` environment variable to 1. So you could conditionally apply some flags in `setup.py`. 

https://cibuildwheel.readthedocs.io/en/stable/faq/#building-packages-with-optional-c-extensions

I know that it is a workaround, not a solution requested by you, but maybe it is enough for your case.

**Reply by andyfaff** [reply]
Im not sure that will work. The build hierarchy is cibuildwheel --> pip-->mesonpy (the build backend in pyproject.toml) --> meson --> ninja. I'm trying to tell meson to do a cross compile. The normal way of specifying a cross-compile is `meson setup --cross-file cross_arm64.txt build`. This tells meson to use `build` as the build directory and place all of its build configuration in there. I can therefore do something like:
```
meson setup --cross-file cross_arm64.txt build
python -m build -Cbuilddir=build
```
and the build frontend will make a cross platform wheel. However, if using cibuildwheel I don't get to issue the second command, so I need cibuildwheel to pass in that extra configuration for me. The project I'm using doesn't use setup.py in the meson build.

**Comment by oscarbenjamin** [comment, marked as answer]
This is now possible with `CIBW_CONFIG_SETTINGS`:
https://cibuildwheel.pypa.io/en/stable/options/#config-settings

**Reply by henryiii** [reply]
There are two things: the build backend settings, which are `-C` flags, are possible via config-settings, yes. For arbitrary build frontend flags, that's also possible now with a custom syntax when setting build-frontend (table in TOML, semicolon in environment variable).
